### PR TITLE
net/devif: add common funtion to build L2 headers before sending

### PR DESCRIPTION
**Pull request by anchao**
## Summary

net/devif: add common funtion to build L2 headers before sending
net/devif: move preprocess of txpoll into common code

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: net/devif/devif_poll.c [hunk 1]
```diff
@@ -181,6 +181,63 @@ static void devif_packet_conversion(FAR struct net_driver_s *dev,
...
+ * Assumptions:
+ *   This function is called from the MAC device driver with the network
+ *   locked.
+ *
+ ****************************************************************************/
+
+static int devif_out(FAR struct net_driver_s *dev,
+                     devif_poll_callback_t callback)
+{
+  FAR struct iob_s *frag;
```
Comment (xiaoxiang781216) [inline net/devif/devif_poll.c:200]: not use?

File: net/devif/devif_poll.c [hunk 2]
```diff
@@ -181,6 +181,63 @@ static void devif_packet_conversion(FAR struct net_driver_s *dev,
...
+  int bstop;
+
+  if (dev->d_len == 0)
+    {
+      return 0;
+    }
+
+  bstop = devif_loopback(dev);
```
Comment (xiaoxiang781216) [inline net/devif/devif_poll.c:208]: move to next patch

Comment (anchao) [reply net/devif/devif_poll.c:208]: Done

File: net/devif/devif_poll.c (hunk 1 above)
Comment (anchao) [reply net/devif/devif_poll.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

File: arch/arm/src/c5471/c5471_ethernet.c [hunk 3]
```diff
@@ -1015,51 +1015,19 @@ static int c5471_txpoll(struct net_driver_s *dev)
...
-#ifdef CONFIG_NET_IPv4
-      else
-#endif
-        {
-          neighbor_out(&priv->c_dev);
-        }
-#endif /* CONFIG_NET_IPv6 */
+  /* Send the packet */
 
-      if (!devif_loopback(&priv->c_dev))
```
Comment (xiaoxiang781216) [inline arch/arm/src/c5471/c5471_ethernet.c:1046]: let's move devif_loopback to net/devif/devif.h

Comment (anchao) [reply arch/arm/src/c5471/c5471_ethernet.c:1046]: Done

File: include/nuttx/net/netdev.h [hunk 4]
```diff
@@ -564,6 +564,20 @@ int sixlowpan_input(FAR struct radio_driver_s *ieee,
...
+ * Description:
+ *   Generic callback before device output to build L2 headers before sending
+ *
+ * Assumptions:
+ *   This function is called from the MAC device driver with the network
+ *   locked.
+ *
+ ****************************************************************************/
+
+int devif_out(FAR struct net_driver_s *dev, devif_poll_callback_t callback);
```
Comment (xiaoxiang781216) [inline include/nuttx/net/netdev.h:579]: let's move to devif/devif.h

File: drivers/usbhost/usbhost_cdcmbim.c [hunk 5]
```diff
@@ -2279,12 +2279,9 @@ static int cdcmbim_txpoll(struct net_driver_s *dev)
 
   if (priv->netdev.d_len > 0)
```
Comment (xiaoxiang781216) [inline drivers/usbhost/usbhost_cdcmbim.c:2280]: remove the check

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**